src/models.js

```diff
@@ -142,20 +144,27 @@ async function constructSession(pretrained_model_name_or_path, fileName, options
 /**
  * Validate model inputs
  * @param {InferenceSession} session The InferenceSession object that will be run.
- * @param {Object} inputs The inputs to check.
+ * @param {Record<string, Tensor>} inputs The inputs to check.
  * @returns {Promise<Object>} A Promise that resolves to the checked inputs.
  * @throws {Error} If any inputs are missing.
  * @private
  */
 async function validateInputs(session, inputs) {
-    // NOTE: Only create a shallow copy
+    // NOTE: Create either a shallow or deep copy based on `onnx.wasm.proxy`
     const checkedInputs = {};
```

Comment:
```suggestion
    const checkedInputs = Object.create(null);
```
Another thing I've noticed, there could be a rare case where one of the model's input names corresponds to a built-in object name (e.g., `toString`), which would cause the later `(!tensor)` check to fail. Probably good to fix it now too.

Reply:
Hm, that's interesting, normally a `Map` is used then, but `ONNXRuntime` only supports an object...

Given this code and putting "extern" in front of `validateInputs` for testing:

```js
const session = {
    inputNames: ['a', 'b', 'toString', 'valueOf']
};
const inputs = {
    a: new Tensor("float32", [1, 2, 3, 4], [2, 2]),
    b: new Tensor("float32", [1, 2, 3, 4], [1, 4]),
    //toString: new Tensor("float32", [1, 2, 3, 4], [1, 4]),
    valueOf: new Tensor("float32", [1, 2, 3, 4], [4, 1]),
};
const checkedInputs = validateInputs(session, inputs);
console.log(Object.keys(checkedInputs));
```

I tested two scenarios:

![image](https://github.com/xenova/transformers.js/assets/5236548/1b4f0bce-ff1d-40e3-acab-6e1d0044a897)

This happens with or without the `Object.create(null)` change, because `tensor` comes from the `inputs` arguments.

This should probably throw, but it doesn't:

```js
env.backends.onnx.wasm.proxy = false;
const session = {
    inputNames: ['toString', 'valueOf']
};
const inputs = {};
const checkedInputs = validateInputs(session, inputs);
console.log(Object.keys(checkedInputs));
```

So I just tested a `tensor instanceof Tensor` check and that detects the missing inputs:

![image](https://github.com/xenova/transformers.js/assets/5236548/d408fbe7-af85-4588-93ba-4d6c815b0c8a)

Thank you for your quick input and review!